ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGSchema.scala

```diff
@@ -288,16 +287,20 @@ private[postgresql] object PGSchema {
     ): Field[FROM, Option[Instant], _] =
       PGTimestampOptional(extractor)
 
-    override def intOptional[FROM, _](extractor: FROM => Option[Int]): Field[FROM, Option[Int], _] =
-      PGIntOptional(extractor)
+    override def insert[FROM](tableName: String)(
+        fields: (String, Field[FROM, _, _])*
+    ): Table[FROM] =
+      PGTable.transposedInsert(tableName)(fields: _*)
 
-    override def boolean[FROM, _](extractor: FROM => Boolean): Field[FROM, Boolean, _] =
-      PGBoolean(extractor)
+    override def delete[FROM](tableName: String)(field: (String, Field[FROM, _, _])): Table[FROM] =
+      PGTable.transposedDelete(tableName)(field)
 
-    override def booleanOptional[FROM, _](
-        extractor: FROM => Option[Boolean]
-    ): Field[FROM, Option[Boolean], _] =
-      PGBooleanOptional(extractor)
+    override def packageInsert(
+        tableName: String
+    )(fields: (String, Field[DbDto.Package, _, _])*): Table[DbDto.Package] =
+      PGTable.transposedInsertWithSuffix(tableName, "on conflict (package_id) do nothing")(
+        fields: _*
+      )
```

Comment:
Do you think it would make sense to make this more generic? Something like:
```scala
// Assumes the primary key uniquely identifies the content of the row.
// Conflicts on the primary key are safely and silently ignored.
def idempotentInsert(tableName: String, primaryKey: String)(...) =
  PGTable.transposedInsertWithSuffix(tableName, s"on conflict ($primaryKey) do nothing")
```

Reply:
good idea, will do